contracts/smartdeploy/src/registry/contract.rs

```diff
@@ -32,9 +31,29 @@ pub struct DeployEventData {
     deployer: Address,
     contract_id: Address,
 }
+#[contracttype(export = false)]
+pub struct ContractRegistry(pub Map<String, ContractState>);
 
 #[contracttype(export = false)]
-pub struct ContractRegistry(pub Map<String, Address>);
+#[derive(Clone)]
+pub enum ContractState {
+    ContractId(Address),
+    Owner(Address, Address)
+}
+
+impl ContractState {
+    pub fn contract_id(&self) -> &Address {
+        match self {
+            Self::ContractId(id) | Self::Owner(id, _) => id,
```

Comment:
This looks like you need to run `cargo fmt`

Reply:
`cargo fmt` doesn't change that function...